modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -0,0 +1,1122 @@
+        // This returns a local PWLD vector which must still be integrated
+        // against the diffusion test functions.
+        auto Ss = SetNodalDiffusionScatterSource(phi0_old);
+
+        // Assemble the normal terms of the RHS of the diffusion solver,
+        // add in the second moment correction, and solve the system
+        auto b = AssembleDiffusionRHS(Sf + Ss);
+        diffusion_solver_->AddToRHS(b + correction);
+        diffusion_solver_->Solve(phi0, /*use_initial_guess=*/true);
+
+        // Bump the new solution to old
+        phi0_old = phi0;
+      }
+
+      // Compute a new diffusion eigenvalue. This requires mapping the
+      // diffusion solution back to the transport discretization.
+      std::vector<double> phi;
+      TransferDiffusionToTransport(phi0, phi);
+      double F0 = lbs_solver_.ComputeFissionProduction(phi);
+      lambda = F0 / F0_old * lambda_old;
+
+      // Check for convergence
+      double lambda_change = std::fabs(lambda / lambda_old - 1.0);
+      if (accel_pi_verbose_)
+        log.Log() << "SMM PI iteration " << m << "  lambda " << lambda << "  change "
+                  << lambda_change << (lambda_change < accel_pi_k_tol_ ? "  CONVERGED" : "");
+
+      // Bump variables for next iteration
+      lambda_old = lambda;
+      F0_old = F0;
+      phi0_m = phi0;
+
+      if (lambda_change < accel_pi_k_tol_)
+        break;
+
+    } // acceleration
+
+    // Project back to transport and check convergence
+    k_eff_ = lambda;
+    k_eff_change = std::fabs(k_eff_ / k_eff_ell - 1.0);
+    k_eff_ell = k_eff_;
+
+    TransferDiffusionToTransport(phi0, phi_new_local_);
+
+    const double production = lbs_solver_.ComputeFissionProduction(phi_new_local_);
+    lbs_solver_.ScalePhiVector(PhiSTLOption::PHI_NEW, lambda / production);
+
+    const auto phi_change = CheckScalarFluxConvergence(phi_new_local_, phi_ell);
+    lbs_solver_.GSScopedCopyPrimarySTLvectors(front_gs_, phi_new_local_, phi_old_local_);
+    lbs_solver_.GSScopedCopyPrimarySTLvectors(front_gs_, phi_new_local_, phi_ell);
+    ++nit;
+
+    if (lbs_solver_.Options().verbose_outer_iterations)
+    {
+      std::stringstream ss;
+      ss << program_timer.GetTimeString() << " "
+         << "  Iteration " << std::setw(5) << nit << "  k_eff " << std::setw(14)
+         << std::setprecision(10) << k_eff_ << "  k_eff change " << std::setw(12) << k_eff_change
+         << "  phi change " << std::setw(12) << phi_change
+         << (k_eff_change < k_tolerance_ ? " CONVERGED" : "");
+      log.Log() << ss.str();
+    }
+
+    if (k_eff_change < k_tolerance_)
+      break;
+  }
+
+  // Print summary
+  log.Log() << "\n";
+  log.Log() << "        Final k-eigenvalue    :        " << std::setprecision(7) << k_eff_;
+  log.Log() << "        Final change          :        " << std::setprecision(6) << k_eff_change
+            << " (num_TrOps:" << front_wgs_context_->counter_applications_of_inv_op_ << ")"
```

Comment:
Should `num_TrOps` be something a user would understand better? It seems like it was a variable name, but it is not any more?

Reply:
I agree it is not very clear for an average user. I think it stands for number of transport operations (or action of the transport operator).  Myself, I am not 100% sure if a unit of such `TrOps` is a sweep per group and per direction or if it is some large unit of work.

Reply:
It is the number of sweeps (transport operations) done for a given groupset. This is recycled from other existing outputs in the code. I can change this to something else, just let me know what you suggest.

Reply:
So `Number of sweeps:` ?

Reply:
Changing to `Sweep Count:`